financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/util/IdentityCommonUtils.java

```diff
@@ -700,4 +710,117 @@ public static List<RequestedClaim> getRequestedClaims(String token) throws Ident
+            ConsentManagementException {
+
+        FinancialServicesConfigParser configParser = FinancialServicesConfigParser.getInstance();
+        if (!configParser.isPreInitiatedConsent()) {
+            String commonAuthId = getCommonAuthId(oauthAuthzMsgCtx);
+            return getConsentIdFromCommonAuthId(commonAuthId);
+        }
+
+        if ("requestObject".equals(configParser.getAuthFlowConsentIdSource())) {
+            return getConsentIdFromRequestObject(oauthAuthzMsgCtx.getAuthorizationReqDTO());
+        }
+
+        if ("requestParam".equals(configParser.getAuthFlowConsentIdSource())) {
+            return getConsentIdFromRequestParam(oauthAuthzMsgCtx.getAuthorizationReqDTO());
+        }
+
+        return null;
+    }
+
+    private static String getConsentIdFromCommonAuthId(String commonAuthId) throws ConsentManagementException {
+
+        return new ConsentCoreServiceImpl()
+                .getConsentIdByConsentAttributeNameAndValue(IdentityCommonConstants.COMMON_AUTH_ID, commonAuthId)
+                .get(0);
+    }
+
+    @SuppressFBWarnings("COOKIE_USAGE")
+    private static String getCommonAuthId(OAuthAuthzReqMessageContext oAuthAuthzReqMessageContext) {
+
+        Cookie[] cookies = oAuthAuthzReqMessageContext.getAuthorizationReqDTO().getCookie();
+        String commonAuthId = StringUtils.EMPTY;
+        ArrayList<Cookie> cookieList = new ArrayList<>(Arrays.asList(cookies));
+        for (Cookie cookie : cookieList) {
+            if (IdentityCommonConstants.COMMON_AUTH_ID.equals(cookie.getName())) {
+                commonAuthId = cookie.getValue();
+                break;
+            }
+        }
+        return commonAuthId;
+    }
+
+    private static String getConsentIdFromRequestObject(OAuth2AuthorizeReqDTO oAuth2AuthorizeReqDTO)
+            throws JsonProcessingException {
+
+        FinancialServicesConfigParser configParser = FinancialServicesConfigParser.getInstance();
+        String json = oAuth2AuthorizeReqDTO.getEssentialClaims();
+        String jsonPath = configParser.getConsentIdExtractionJsonPath();
+
+        if (json == null || json.isEmpty() || jsonPath == null || jsonPath.isEmpty()) {
+            return null; // Return null if input is invalid
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode node = mapper.readTree(json);
+        JsonNode targetNode = node.at(jsonPath);
+        return extractFromRegex(targetNode.asText());
+    }
+
+    private static String getConsentIdFromRequestParam(OAuth2AuthorizeReqDTO oAuth2AuthorizeReqDTO) {
+
+        // TODO: need to support other request parameters based on the requirements
+        StringBuilder scopesString = new StringBuilder();
+        for (String scope : oAuth2AuthorizeReqDTO.getScopes()) {
+            scopesString.append(scope).append(" ");
+        }
+        return extractFromRegex(scopesString.toString().trim());
+    }
+
+    private static String extractFromRegex(String text) {
```

Comment:
Method name indicates that this is general. But the pattern is for consent id.

Reply:
Fixed with 533d2b6e24e4c38d24460cd3042132849e579b08